homeassistant/components/google_generative_ai_conversation/conversation.py

```diff
@@ -164,6 +164,10 @@ def __init__(self, entry: ConfigEntry) -> None:
             model="Generative AI",
             entry_type=dr.DeviceEntryType.SERVICE,
         )
+        if self.entry.options.get(CONF_LLM_HASS_API):
```

Comment:
Call _async_entry_update_listener to avoid duplicate code? Or create a new function that you call in both places?

BTW, I have noticed that if you rename the config entry the entity doesn't get renamed until you reload the config entry. Does the update listener you added get called when an entry is renamed? If yes, move the device info in that function too? Or maybe just have the update listener call the __init__?

Reply:
device info is only processed when we re-add the entity. So in that case, we would need to reload the config entry.

Reply:
Updated to reload config entry.